ehcache-core/src/main/java/org/ehcache/core/EhcacheBase.java

```diff
@@ -68,7 +69,7 @@
 
   protected final Logger logger = EhcachePrefixLoggerFactory.getLogger(EhcacheBase.class);
 
-  protected final StatusTransitioner statusTransitioner = new StatusTransitioner();
+  protected final StatusTransitioner statusTransitioner;
```

Comment:
You identified the problem, but instead of solving it by restoring the injection of the `StatusTransitioner`, why not just have the `PersistentUserManagedEhcache` drop its instance of the transitioner and directly reference the one in Ehcache? That way you don't need to reintroduce the constructor parameter?

Reply:
I have already tried for all possibilities to not propagate this - `statusTransitioner` object from `PersistentUserManagedEhcache` to `EhcacheBase`.
The `statusTransitioner` attribute in `PersistentUserManagedEhcache `class is been involved in other life cycle methods - `create(), destroy(), init(), close()` .. so on
So `EhcacheBase `class needs the same copy of `statusTransitioner` object, else cache is not been properly initialized and `cahce.put(-,-)` is failing with - `StoreAccessException`

Reply:
Updated the fix, based on our discussion. validated the junits